SecretsOTheP/EQMacEmu
# Feature/add hardcore ghost spawn command

Prototype for being able to spawn & inspect npc representations of hardcore players.

This is for use in the Hardcore Hall of Fame. https://discord.com/channels/1133452007412334643/1301955243927408801

Testing steps:

Copy some player data into the hardcore ghosts tables (see .sql files for example)

Enter a zone and use command #spawnhardcoreghost playername

That player's ghost will appear where you're standing & have all of their items equipped.  You can right click to inspect them to see what items they have.

Next steps:

- Preserve hardcore player's important data when they die so we can reproduce them in the hall of fame
- Add support for spawning specific players via command from existing player data if they are still alive
- Set up an empty zone (high keep) to display all of the characters in
- Set up spawn positions for each class to show the players in different areas of the zone
- Figure out how to dynamically assign those spawn positions to players from the backed up hardcore data & live player data based off of a leaderboard style (highest level & first to achieve gets the most prominent positions)
- Set up way to port to this zone in EC tunnel (npc)
- Set up top 3 hardcore players as avatars in EC tunnel near the npc as a reward
- Add quests to the hardcore ghosts so people can ask them about their journey.
- Add support for players of those hardcore ghosts to update the data about themselves in the DB automatically
- Automatically track certain data about players during their journey like favorite zone, favorite npc to kill, most difficult zones entered, etchardcore players as avatars in EC tunnel near the npc as a reward
- Add quests to the hardcore ghosts so people can ask them about their journey.
- Add support for players of those hardcore ghosts to update the data about themselves in the DB automatically
- Automatically track certain data about players during their journey like favorite zone, favorite npc to kill, most difficult zones entered, etc
